### PR TITLE
Wrap, doc, test GitBlame

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]

File: base/libgit2/blame.jl [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+function GitBlame(repo::GitRepo, path::AbstractString; options::BlameOptions=BlameOptions())
+    blame_ptr_ptr = Ref{Ptr{Void}}(C_NULL)
+    @check ccall((:git_blame_file, :libgit2), Cint,
+                  (Ptr{Ptr{Void}}, Ptr{Void}, Cstring, Ptr{BlameOptions}),
+                   blame_ptr_ptr, repo.ptr, path, Ref(options))
+    return GitBlame(repo, blame_ptr_ptr[])
+end
+
+function Base.count(blame::GitBlame)
+    return ccall((:git_blame_get_hunk_count, :libgit2), Int32, (Ptr{Void},), blame.ptr)
```
Comment (tkelman) [inline base/libgit2/blame.jl]: how many different measures can a blame object be counted by? this seems too context-less and ambiguous to extend Base.count

File: base/libgit2/blame.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+end
+
+function Base.getindex(blame::GitBlame, i::Integer)
```
Comment (tkelman) [inline base/libgit2/blame.jl]: should these be iterable then?

File: base/libgit2/blame.jl [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    end
+    hunk_ptr = ccall((:git_blame_get_hunk_byindex, :libgit2),
+                      Ptr{BlameHunk},
+                      (Ptr{Void}, Csize_t), blame.ptr, i-1)
+    return unsafe_load(hunk_ptr)
+end
+
+function Base.show(io::IO, blame_hunk::BlameHunk)
+    println(io, "GitBlameHunk:")
+    println(io, "Original path:", blame_hunk.orig.path)
```
Comment (tkelman) [inline base/libgit2/blame.jl]: space after colon?

File: base/libgit2/blame.jl [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    println(io, "Lines in hunk: ", blame_hunk.lines_in_hunk)
+    println(io, "Final commit oid: ", blame_hunk.final_commit_id)
+    print(io, "Final signature: ")
+    show(io, blame_hunk.final_signature)
```
Comment (tkelman) [inline base/libgit2/blame.jl]: will this include a trailing newline?

File: base/libgit2/types.jl [hunk 5]
```diff
@@ -645,6 +661,41 @@ mutable struct Signature
...
+Matches the [`git_blame_hunk`](https://libgit2.github.com/libgit2/#HEAD/type/git_blame_hunk) struct.
+The fields represent:
+    * `lines_in_hunk`: the number of lines in this hunk of the blame
+    * `final_commit_id`: the OID of the commit where this section was last changed
+    * `final_start_line_number`: the *one based* line number in the file where the
+       hunk starts, in the *final* version of the file
+    * `final_signature`: the [`GitSignature`] of the person who last modified this hunk
+    * `orig_commit_id`: the OID of the commit where this hunk was first found
+    * `orig_path`: the path to the file where the hunk originated. This may be different
+       than the current/final path, for instance if the file has been moved.
```
Comment (tkelman) [inline base/libgit2/types.jl]: little odd that this is the only bullet with sentences

File: base/libgit2/blame.jl (hunk 1 above)
Comment (kshyatt) [inline base/libgit2/blame.jl]: Just one, it's tell you how many "hunks" are in the blame for the file.

File: base/libgit2/blame.jl (hunk 2 above)
Comment (kshyatt) [inline base/libgit2/blame.jl]: you mean with `next` and `done`? I guess so...

File: base/libgit2/blame.jl (hunk 3 above)
Comment (kshyatt) [inline base/libgit2/blame.jl]: Good catch, I've been writing too much Python recently...

File: base/libgit2/blame.jl (hunk 1 above)
Comment (tkelman) [inline base/libgit2/blame.jl]: lines? files? this isn't consistent with the Base.count generic function behavior of telling you the number of true elements of an array or iterator. maybe call it counthunks?

File: base/libgit2/blame.jl [hunk 6]
```diff
@@ -24,12 +24,13 @@ end
...
 function Base.show(io::IO, blame_hunk::BlameHunk)
     println(io, "GitBlameHunk:")
-    println(io, "Original path:", blame_hunk.orig.path)
+    println(io, "Original path: ", blame_hunk.orig.path)
     println(io, "Lines in hunk: ", blame_hunk.lines_in_hunk)
     println(io, "Final commit oid: ", blame_hunk.final_commit_id)
     print(io, "Final signature: ")
     show(io, blame_hunk.final_signature)
     println(io, "Final commit oid: ", blame_hunk.orig_commit_id)
     print(io, "Final signature: ")
```
Comment (tkelman) [inline base/libgit2/blame.jl]: Original ?

File: base/libgit2/blame.jl (hunk 1 above)
Comment (kshyatt) [inline base/libgit2/blame.jl]: I like `counthunks`, yeah :)

File: base/libgit2/blame.jl (hunk 6 above)
Comment (kshyatt) [inline base/libgit2/blame.jl]: you mean on the commit id or something else?

Comment (tkelman) [inline base/libgit2/blame.jl]: these lines are showing orig_commit_id and orig_signature

File: base/libgit2/blame.jl (hunk 4 above)
Comment (tkelman) [inline base/libgit2/blame.jl]: will there need to be a `println()` before the `Original commit oid` line ?

File: base/libgit2/blame.jl (hunk 6 above)
Comment (kshyatt) [inline base/libgit2/blame.jl]: Fixed :)

Comment (kshyatt) [inline base/libgit2/blame.jl]: GitHub was highlighting the wrong thing again, sorry

**Comment by kshyatt** [comment]
@tkelman anything else I should do, or can we squash?

File: base/libgit2/types.jl [hunk 7]
```diff
@@ -645,6 +661,43 @@ mutable struct Signature
...
+    * `final_signature`: the signature of the person who last modified this hunk. You will
+       need to pass this to [`Signature`](@ref) to access its fields.
+    * `orig_commit_id`: the [`GitHash`](@ref) of the commit where this hunk was first found.
+    * `orig_path`: the path to the file where the hunk originated. This may be different
+       than the current/final path, for instance if the file has been moved.
+    * `orig_start_line_number`: the *one based* line number in the file where the
+       hunk starts, in the *original* version of the file at `orig_path`.
+    * `orig_signature`: the signature of the person who introduced this hunk. You will
+       need to pass this to [`Signature`](@ref) to access its fields.
+    * `boundary`: `1` if the original commit is a "boundary" commit (for instance, if it's
```
Comment (tkelman) [inline base/libgit2/types.jl]: the char 1?

Comment (kshyatt) [inline base/libgit2/types.jl]: That's what it says in the libgit2 docs ¯\_(ツ)_/¯

Comment (tkelman) [inline base/libgit2/types.jl]: so it should be `'1'` in julia syntax then?

**Review by ararslan — APPROVED** [review]
Codegen test failures are unrelated but _really_ weird

**Pull request merged**